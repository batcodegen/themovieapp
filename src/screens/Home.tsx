import {useTheme} from '@react-navigation/native';
import React, {useCallback} from 'react';
import {FlatList, SafeAreaView, StyleSheet, Text, View} from 'react-native';
import {Movie} from '../api/popularmovie';
import Header from '../components/Header';
import MovieCard from '../components/MovieCard';
import {useChangeLanguage} from '../hooks/useChangeLanguage';
import {useGetPopularMovies} from '../hooks/useGetPopularMovies';
import DrawerWrapper from '../navigation/DrawerWrapper';
import {useTranslation} from 'react-i18next';

const Home = () => {
  const {colors} = useTheme();
  const {t} = useTranslation();
  useChangeLanguage();
  const {data, error, fetchNextPage, isLoadingMoreMovies} =
    useGetPopularMovies();

  const renderMovieCard = ({item}: {item: Movie}) => {
    return (
      <MovieCard
        id={item.id.toString()}
        image={item.poster_path}
        rating={item.vote_average}
        releaseDate={item.release_date}
        title={item.original_title}
        voteCount={item.vote_count}
      />
    );
  };

  const renderFooter = useCallback(() => {
    if (isLoadingMoreMovies) {
      return <Text style={styles.loadingMore}>Loading More...</Text>;
    }
    return <View />;
  }, [isLoadingMoreMovies]);

  if (error) {
    return (
      <View>
        <Text>Error</Text>
      </View>
    );
  }

  return (
    <DrawerWrapper>
      <SafeAreaView
        style={[styles.container, {backgroundColor: colors.background}]}>
        <Header screenName={t('home.popularMovies')} />
        <FlatList
          data={data}
          numColumns={2}
          keyExtractor={item => item.id.toString()}
          renderItem={renderMovieCard}
          showsVerticalScrollIndicator={false}
          maxToRenderPerBatch={10}
          bounces={false}
          initialNumToRender={20}
          removeClippedSubviews={true}
          onEndReached={fetchNextPage}
          onEndReachedThreshold={0.5}
          ListFooterComponent={renderFooter}
        />
      </SafeAreaView>
    </DrawerWrapper>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {flex: 1},
  loadingMore: {textAlign: 'center'},
});
