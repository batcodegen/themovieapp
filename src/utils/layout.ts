import {I18nManager} from 'react-native';

const isRTL: boolean = I18nManager.isRTL;
const textinputAlignOnRTL: 'right' | 'left' = isRTL ? 'right' : 'left';
const currentLanguageLocale: string = isRTL ? 'ar-AE' : 'en-US';

export {isRTL, textinputAlignOnRTL, currentLanguageLocale};
