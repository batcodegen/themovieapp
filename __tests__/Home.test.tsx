import React from 'react';
import {screen} from '@testing-library/react-native';
import Home from '../src/screens/Home';
import {renderWithProviders} from './utils/test-utils';

describe('Home component', () => {
  test('Renders Home correctly', async () => {
    renderWithProviders(<Home />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
