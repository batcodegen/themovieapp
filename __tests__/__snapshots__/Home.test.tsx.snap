// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders Home correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "borderRadius": undefined,
        "overflow": "hidden",
        "transform": [
          {
            "scale": undefined,
          },
        ],
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      [
        {
          "flex": 1,
        },
        {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "paddingHorizontal": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "flex-start",
            "flex": 0.2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 30,
              },
              undefined,
              {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          îµ’
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 0.6,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 18,
                "fontWeight": "bold",
                "letterSpacing": 0.2,
              },
              {
                "color": undefined,
              },
            ]
          }
        >
          home.popularMovies
        </Text>
      </View>
      <View
        style={
          {
            "flex": 0.2,
          }
        }
      />
    </View>
    <RCTScrollView
      ListFooterComponent={[Function]}
      bounces={false}
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      initialNumToRender={20}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={true}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View />
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
